lib/thrift/protocol/binary.ex

```diff
@@ -111,7 +108,7 @@ defmodule Thrift.Protocol.Binary do
   end
 
   def deserialize(:application_exception, binary) when is_binary(binary) do
-    do_read_application_exception(binary, %TApplicationException{})
+    do_read_application_exception(binary, TApplicationException.exception(type: :unknown))
```

Comment:
I think this would be cleaner to accumulate keyword and then pass that to `TApplicationException.exception/1` on `@stop`.